### ISSUE TITLE
How to prevent rust-analyzer importing duplicate or wrong items?

### ISSUE THREAD

**Issue by Ali-Javanmardi** [rust-lang/rust-analyzer#17132]
Hi,

I'm using rust-analyzer in neovim (lunarvim) and it works almost good.
But in some cases when I already have some imported struct or enum from my own crates or libs ( for example customized Result or Error ) rust-analyzer auto completion adds unwanted and duplicated struct or enum from other cartes like serde or std.

every time it happens I face unexpected error in my file and when I check on top of my file I find some unexpected  use lines added there.

Sometime it is really annoying as it reaped several times during editing code.

So please help me find how I can prevent rust-analyzer  from this action or at least prevent duplicate same name import?

**Comment by Natr1x** [comment]
Unsure if you are still having issues with this but in case someone else comes across it.

You should be able to disable the auto import with `rust-analyzer.completion.autoimport.enable = false`.

But if you are new to neovim or unfamiliar with how your plugins interact with the language server protocol it can be tricky to figure out where to set that.

If you are using 'neovim/nvim-lspconfig' I think you would do it when you call setup in the settings:

```lua
local lspconfig = require('lspconfig')
lspconfig.rust_analyzer.setup {
  -- Server-specific settings. See `:help lspconfig-setup`
  settings = {
    ['rust-analyzer'] = {
      completion = {
        autoimport = {
          enable = false,
        },
      },
    },
  },
}
```

However if you are using 'mrcjkb/rustaceanvim' then you should specify configurations there instead:
```lua
vim.g.rustaceanvim = function ()
  return {
    -- LSP configuration
    server = {
      on_attach = set_lsp_keybindings,
      default_settings = {
        -- rust-analyzer language server configuration
        ['rust-analyzer'] = {
          completion = {
            autoimport = {
              enable = false,
            },
          },
        },
      },
    },
  }
end
```
